test/integration/dockerapi_fetcher_test.go

```diff
@@ -28,10 +29,17 @@ func TestDockerAPIHelpers(t *testing.T) {
 	require.NoError(t, err)
 	logAsJSON(t, "Inspect data", &inspectData)
 
-	// fetch stats data
+	// Let stress container have some stresful moments before fetching data.
+	time.Sleep(time.Second * 10)
```

Comment:
Did you consider using [testify's Eventually](https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually), in the same way it is done "manually" in [other tests](https://github.com/newrelic/nri-docker/blob/4bd5599ca69d38f2a08c60e86bd4e8e72d2bd3a6/test/integration/metrics_test.go#L61)

Reply:
I got rid of the "manual" implemention within the refactor.